app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -140,14 +140,22 @@ def pulp3_enabled?
       end
 
       def pulp3_support?(repository)
-        type = Katello::RepositoryTypeManager.repository_types[repository.content_type]
-        type.pulp3_plugin.present? && pulp3_enabled?
+        pulp3_repository_type_support?(repository.content_type)
       end
 
-      def content_pulp3_support?(content_type)
-        content_type = content_type.is_a?(String) ? content_type : content_type.model_class::CONTENT_TYPE
-        type = Katello::RepositoryTypeManager.find_repository_type content_type
-        type.pulp3_plugin && SmartProxy.pulp_master!.capabilities(PULP3_FEATURE).try(:include?, type.pulp3_plugin)
+      def pulp3_repository_type_support?(repository_type)
+        repository_type_obj = repository_type.is_a?(String) ? Katello::RepositoryTypeManager.repository_types[repository_type] : repository_type
```

Comment:
Is this better
```repositoy_type =  Katello::RepositoryTypeManager.repository_types[repository_type] if repository_type.is_a?(String)```

Reply:
I originally had that, but consider the case where a repository_type string is passed in of an invalid type.  repository_type is set to nil and the line below can't print the repository_type